util-store/src/main/java/com/hpe/caf/util/store/StoreUtil.java

```diff
@@ -0,0 +1,95 @@
+     * and the absolute reference of the final location will be returned along with a SHA-1 hash of the data.
+     * @param dataStore the DataStore into which the data will be stored
+     * @param dataStream the stream of data which will be read and put into the DataStore
+     * @param partialReference the partial reference, which the data will be stored relative to
+     * @return absolute reference to the stored data, which can be used to retrieve, plus a SHA-1 hash of the stored data
+     * @throws DataStoreException if the data store cannot service the request
+     */
+    public static HashStoreResult hashStore(final DataStore dataStore, final InputStream dataStream, final String partialReference)
+        throws DataStoreException
+    {
+        final HashingInputStream hashStream = new HashingInputStream(Hashing.sha1(), new BufferedInputStream(dataStream));
+        final String reference = dataStore.store(hashStream, partialReference);
+        final String hash = Hex.encodeHexString(hashStream.hash().asBytes());
+        return new HashStoreResult(reference, hash);
+    }
+
+
+    /**
+     * Store data from a byte array. The data will be stored in the supplied DataStore relative to the partial
+     * reference supplied, and the absolute reference of the final location will be returned
+     * along with a SHA-1 hash of the data.
+     * @param dataStore the DataStore into which the data will be stored
+     * @param data the raw byte data to store
+     * @param partialReference the partial reference, which the data will be stored relative to
+     * @return absolute reference to the stored data, which can be used to retrieve, plus a SHA-1 hash of the stored data
+     * @throws DataStoreException if the data store cannot service the request
+     */
+    public static HashStoreResult hashStore(final DataStore dataStore, final byte[] data, final String partialReference)
+        throws DataStoreException
+    {
+        return new HashStoreResult(dataStore.store(data, partialReference), DigestUtils.sha1Hex(data));
+    }
+
+
+    /**
+     * Store data from a local file. The data will be stored in the supplied DataStore relative to the partial
+     * reference supplied, and the absolute reference of the final location will be returned
+     * along with a SHA-1 hash of the data.
+     * @param dataStore the DataStore into which the data will be stored
+     * @param dataPath path to a file on the local filesystem to store on the remote DataStore
+     * @param partialReference the partial reference, which the data will be stored relative to
+     * @return absolute reference to the stored data, which can be used to retrieve, plus a SHA-1 hash of the stored data
+     * @throws DataStoreException if the data store cannot service the request
+     */
+    public static HashStoreResult hashStore(final DataStore dataStore, final Path dataPath, final String partialReference)
```

Comment:
This method is reading the file twice.  If it just used the first hashStore() overload that takes a stream it would only be reading the file once.